### PR TITLE
fix ColorItem type conversion

### LINKED ISSUE #2399: Color controls for HSB seem to be independant.
It seems that the 3 controls for color in PaperUI are independant controls?  If I slide the Sat, then I get a saturation, but H and B are 0. Likewise for others - if I change Hue, then S and B are 0.

For example, below I moved the brightness, and I get the following -:
```
2016-11-01 17:43:43.490 [DEBUG] [.c.thing.internal.ThingManager:458  ] - Delegating update '0,0,29' for item 'zwave_device_15785c713e3_node15_color_color' to handler for channel 'zwave:device:15785c713e3:node15:color_color'
```

The controls look as follows -:

image

I would have expected saturation and hue to be values as indicated in the image.

### PR DESCRIPTION
**Pull request by sjka**
fixes #2399
Signed-off-by: Simon Kaufmann <simon.kfm@googlemail.com>

### CODE REVIEW

File: bundles/core/org.eclipse.smarthome.core.test/src/test/java/org/eclipse/smarthome/core/library/items/ColorItemTest.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        ColorItem item = new ColorItem("test");
+        item.setState(new PercentType(50));
+        assertEquals(new HSBType(new DecimalType(0), new PercentType(0), new PercentType(50)), item.getState());
+    }
+
+    @Test
+    public void testSetStateWithDecimalType() {
+        ColorItem item = new ColorItem("test");
+        item.setState(new DecimalType(0.5));
+        assertEquals(new HSBType(new DecimalType(0), new PercentType(0), new PercentType(50)), item.getState());
```
Comment (kaikreuzer) [inline bundles/core/org.eclipse.smarthome.core.test/src/test/java/org/eclipse/smarthome/core/library/items/ColorItemTest.java]: DecimalType is NOT an accepted state - I would therefore expect that it is ignored or an exception is thrown.

Do we currently have situations in the code where a ColorItem has to deal with DecimalTypes?

File: bundles/core/org.eclipse.smarthome.core.test/src/test/java/org/eclipse/smarthome/core/library/items/ColorItemTest.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        item.setState(new PercentType(50));
+        assertEquals(new HSBType(new DecimalType(75), new PercentType(75), new PercentType(50)), item.getState());
+    }
+
+    @Test
+    public void testUpdateStateWithDecimalType() {
+        ColorItem item = new ColorItem("test");
+        item.setState(new HSBType(new DecimalType(75), new PercentType(75), new PercentType(75)));
+
+        item.setState(new DecimalType(0.5));
```
Comment (kaikreuzer) [inline bundles/core/org.eclipse.smarthome.core.test/src/test/java/org/eclipse/smarthome/core/library/items/ColorItemTest.java]: see above

File: bundles/core/org.eclipse.smarthome.core.test/src/test/java/org/eclipse/smarthome/core/library/items/ColorItemTest.java (hunk 1 above)
Comment (sjka) [inline bundles/core/org.eclipse.smarthome.core.test/src/test/java/org/eclipse/smarthome/core/library/items/ColorItemTest.java]: > DecimalType is NOT an accepted state
okay, that's correct, I'll remove it.

> I would therefore expect that it is ignored or an exception is thrown
However, the check is not (and never was) done in the item itself. So far, if bogus was sent, the item was updated with it. I can enhance the ThingManager to really ignore a channel change and not update the item in such a case.

**Comment by kaikreuzer** [comment]
Many thanks!

**Pull request merged**